atomic_reactor/build.py

```diff
@@ -178,21 +183,56 @@ def set_df_path(self, path):
         for image in dfp.parent_images:
             self.parent_images[image] = None
 
-    def set_base_image(self, base_image):
+    def set_base_image(self, base_image, pulled=True, insecure=False):
         logger.info("setting base image to '%s'", base_image)
         self.base_image = ImageName.parse(base_image)
         self.original_base_image = self.original_base_image or self.base_image
         self.parent_images[str(self.original_base_image)] = base_image
+        self._base_image_pulled = pulled
+        self._base_image_insecure = insecure
 
-    def inspect_base_image(self):
+    # inspect base image lazily just before it's needed - pre plugins may change the base image
+    @property
+    def base_image_inspect(self):
         """
         inspect base image
 
         :return: dict
         """
-        logger.info("inspecting base image '%s'", self.base_image)
-        inspect_data = self.tasker.inspect_image(self.base_image)
-        return inspect_data
+        if self._base_image_inspect is None:
+            if not self._base_image_pulled:
+                self._base_image_inspect = get_inspect_for_image(self.base_image,
+                                                                 self.base_image.registry,
+                                                                 self._base_image_insecure)
+            else:
+                try:
+                    self._base_image_inspect = self.tasker.inspect_image(self.base_image)
+
+                except docker.errors.NotFound:
+                    # If the base image cannot be found throw KeyError -
```

Comment:
_base_image_pulled is explicitly set only in pull_base_image via set_base_image, that is also used in different plugins, I know for sure only in pull_base_image that I've pulled that image, so if in worker build by any chance some plugin after pull_base_image (change_from_in_df) would change base image to wrong image, it would fail here